### PR TITLE
[5.8] Use Null coalescing operator to refactor

### PR DESCRIPTION
**Pull request by richardkeep**

### CODE REVIEW

**Review by driesvints — APPROVED** [review]

File: src/Illuminate/Http/Concerns/InteractsWithInput.php [hunk 1]
```diff
@@ -309,9 +309,7 @@ public function allFiles()
     {
         $files = $this->files->all();
 
-        return $this->convertedFiles
-                    ? $this->convertedFiles
-                    : $this->convertedFiles = $this->convertUploadedFiles($files);
+        return $this->convertedFiles = $this->convertedFiles ?? $this->convertUploadedFiles($files);
```
Comment (martinbean) [inline src/Illuminate/Http/Concerns/InteractsWithInput.php]: Could this not be shortened further to just `return $this->convertedFiles ?? $this->convertUploadedFiles($files);`?

Comment (richardkeep) [reply src/Illuminate/Http/Concerns/InteractsWithInput.php]: No, we also want to set the value for $this->convertedFiles when it is falsy before returning.

Comment (martinbean) [reply src/Illuminate/Http/Concerns/InteractsWithInput.php]: @richardkeep Gotcha 🙂

**Comment by rellect** [comment]
This change may have side effects as it's not identical to the original behavior.
The correct operator to use here is the tenary operator:
`return $this->convertedFiles ?: $this->convertedFiles = $this->convertUploadedFiles($files);`

**Comment by driesvints** [comment]
@rellect I think it's the same result in the end, right? Although your solution makes a bit more sense.

**Pull request merged**